### PR TITLE
[Base] New Controls

### LINKED ISSUE #4787: Rework skeleton and volume toolbar
The current tools for skeleton and volume tracings are not very explorable. A lot of behavior is hidden behind shortcuts and or not intuitive modes. For example, to create skeletons you have to be in the "move" tool :see_no_evil: And the existence of the upcoming fill-tool will also be very hard to find out.

There are multiple ways to improve this situation (and we might want to do this in multiple stages).

My vision for a round experience would be:
- have a tool for each dedicated action and use icons instead of labels (e.g.,  the new fill-tool should have an own button)
  - theoretically, actions such as "removing an edge" or "merging two trees" could be a separate tool, too (using the existing shortcuts could still work)
- categorize the tools according to skeleton vs volume. maybe, it could also make sense to have a general skeleton-vs-volume mode button (so, you are either in the skeleton or in the volume mode). I think, this would also be intuitive for users, since one usually focuses on one tracing type (even in hybrids). Additionally, it would save space in the navbar.
- different modes within these tools, should be presented in the navbar as additional buttons (see screenshot below for how Gimp does this)
- in general, all shortcuts which modify the tool behavior, should immediately reflect the state change in the toolbar (also comparable to how gimp does it). For example, if you press a modifier while using the brush tool, the mode button would switch to show that you are overwriting only the current cell.

image

### PR DESCRIPTION
**Pull request by philippotto**
- [x] Skeleton tool
  - [x] legacy mode should behave as before
    - [X] main controls
    - [ ] also support modifiers in non-legacy mode to merge trees etc (necessary? already available via context menu)
  - [x] left click/drag is "smart" and capable of
    - [x] create node
    - [x] select node
    - [x] pan
    - [x] move node
    - [ ] these actions should switch to the skeleton tool?
  - [x] right click opens context menu
- [x] Trace/Brush tool
  - [x] legacy mode should behave as before
  - [x] right click opens context menu
    - [x] add "select cell"
    - [x] add "copy cell id"
    - [ ] create new cell id? (not convinced, since that isn't context-specific)
    - [x] add fill cell (necessary, since we removed ctrl+shift+click as a shortcut for fill-cell)
    - [ ] optional
      - [ ] copy cell from previous slice
  - [x] support erasure
    - [X] add separate erase-tools (for brush and trace)
    - [x] avoid multiple erase-buttons
      - [X] only render the one which relates to the active tool
      - [ ] if the above isn't good enough, use dropdowns or something else ?
     - [x] CTRL+Shift + Leftclick should erase
- [x] double-check modifiers which can act on active tools
  - [x] pressing shift in "erase"-brush should not switch tools
  - [ ] ... ?
- [x] order context menu entries depending on active tool (in skeleton mode, skeleton options should appear first)
- [x] double-check unnecessary renderings
  - [x] toolbar should not rerender on mouse-move
  - [ ] ... ?
- [x] explain legacy-mode via tooltip in settings
- [x] adapt interactive tooltips in toolbar when modifiers are pressed
- [X] adapt shortcut hints in status bar to non-legacy mode
  - [x] adapt shortcut hints in status bar when modifiers are pressed
- [x] pull master in
- [x] migration strategy for legacy mode
  - [x] organization-wide setting?
  - [x] migrate recommended settings
  - [x] migrate existing user settings
- [ ] turn plane/oblique/flight switch into dropdown with icons?

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- abc

### Issues:
- fixes #4787

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
@daniel-wer I think it would make sense if you could do a high-level review of this PR. I'll open another PR which pulls the master in separately, so that you can review a clean diff (I just had some trouble with squashing the way I wanted and think this is the safer route to take).

Code-wise I'm not really satisfied with how the tool handling is done, but I don't see a way to improve this without doing an even further refactoring. The tools work as follows:

- the active tool is stored in the `Store`
- depending on the active tool, the mouse handlers of the corresponding Tool class (see `tool_controls.js`) are delegated
- if a modifier is pressed, the current Tool class is responsible for invoking the action of another tool (so, the Draw tool will delegate to the pick-cell-tool if shift is pressed)
- for the toolbar and the status bar, a tool accessor is used which is able to map from an active tool and pressed modifiers to the "effective tool" (if I press shift with the draw tool, the toolbar/statusbar show the pick-cell-tool as active)
- the "effective tool" is used to get the control-hints in the status bar

It's a bit messy, but as I said: I don't see a good way around this currently.

File: frontend/javascripts/oxalis/model/accessors/tool_accessor.js [hunk 1]
```diff
@@ -193,30 +192,30 @@ export function adaptActiveToolToShortcuts(
...
-    return activeTool;
-  }
-
-  if (activeTool === AnnotationToolEnum.SKELETON) {
-    // The "skeleton" tool is not changed right now (since actions such as moving a node
-    // don't have a dedicated tool). The only exception is "Alt" which switches to the move tool.
-    if (isAltPressed) {
-      return AnnotationToolEnum.MOVE;
+    // These tools do not have any modifier-related behavior currently (except for ALT
+    // which is already handled below)
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/model/accessors/tool_accessor.js:196]: Hm, I don't see how ALT is being handled below for the `ERASE_BRUSH` and `ERASE_TRACE` tools. Isn't the ALT handling still being done in the else branch of this condition (so never hit for these two tools)?

**Review by daniel-wer — COMMENTED** [review]
Lots of great fixes :+1:

File: frontend/javascripts/oxalis/model/accessors/tool_accessor.js (hunk 1 above)
Comment (philippotto) [reply frontend/javascripts/oxalis/model/accessors/tool_accessor.js:196]: It's a bit hidden: https://github.com/scalableminds/webknossos/pull/5422/files/ce5deb14bc2a7090152400ea8d0363d698b174e6..1f6249f3e1913c169bf5ce794565f410dc259cf5#diff-6bae9636ea043e9416c5c662d914cbb44ee20ef5eed61c99d2419472e981b072R222

The logic of this method is a bit convoluted. Ideally, this should probably be also delegated to (and unified with) the tool classes. However, it should work at least :see_no_evil:

**Comment by philippotto** [comment]
I just merged this PR manually into https://github.com/scalableminds/webknossos/pull/5384 which is why I'm closing it here.

**Pull request closed without merging**